src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1869,45 +1887,98 @@ internal virtual void AggregateRenderPackages(IEnumerable<HelixRenderPackage> pa
                         {
                             var range = item.Value;
                             var startIndex = range.start; //Start mesh vertex index
                             var count = range.end - range.start + 1; //Count of mesh vertices
-                            var uniqueId = baseId + ":" + j + MeshKey;
+                            var uniqueId = baseId + ":" + j + MeshKey + "_Instance";
 
                             List<Matrix> instances;
                             if (rp.instanceTransforms.TryGetValue(item.Key, out instances))
                             {
-                                AddMeshData(uniqueId, rp, startIndex, count, drawDead, baseId, rp.Colors,
-                                    rp.ColorsStride, instances);
+                                AddMeshData(uniqueId, m, startIndex, count, drawDead, baseId, rp.Colors,
+                                    rp.ColorsStride, rp.Transform, rp.RequiresPerVertexColoration, instances);
                             }
 
                             //Track cumulative total of mesh vertices added.
-                            meshVertexCountTotal += count;
+                            processedMeshVertexCount += count;
                             j++;
                         }
 
-                        //If all the mesh regions had instance data then we are done with mesh data.
-                        if (meshVertexCountTotal == m.Positions.Count)
+                        //If all the mesh regions had instance data then we are done with mesh data and this Renderpackage.
+                        if (processedMeshVertexCount == m.Positions.Count)
                         {
                             continue;
                         }
 
-                        //Otherwise, clean up the remaining mesh geometry data in the render package to exclude the regions already generated.
-                        var vertexRanges =
-                            new List<(int start, int end)>(rp.MeshVertexRangesAssociatedWithInstancing.Values.ToList());
-
-                        RemoveMeshGeometryByRange(vertexRanges, m);
+                        //Otherwise, add ranges of mesh geometry to exclude for regions already generated related to instancing.
+                        MeshVertexRangesToRemove.AddRange(rp.MeshVertexRangesAssociatedWithInstancing.Values.ToList());
                     }
 
-                    AddMeshData(id, rp, 0, m.Positions.Count, drawDead, baseId, rp.Colors, rp.ColorsStride);
+                    //If mseh vertex ranges have been utilized previously for instantiating instanced geometry or multiple texture maps we only process the remaining mesh data
+                    //We clone the mesh object so that we do not modify the render package data.
+                    if (MeshVertexRangesToRemove.Any())
+                    {
+                        if (mCopy == null)
```

Comment:
why are these null checks necessary?

Reply:
ahh good catch... I was previously working on the copy in a few places.